toolchain/arangoproxy/internal/format/format.go

```diff
@@ -15,16 +15,34 @@ import (
 */
 
 func AdjustCodeForArangosh(code string) string {
-	code = strings.Replace(code, "~", "", -1)
+	out := ""
+
 	if !(strings.Contains(code, "EOFD")) {
 		code = fmt.Sprintf("%s\nprint('EOFD');\n\n\n\n", code)
 	}
-	re := regexp.MustCompile(`(?m)let |const `)
-	code = re.ReplaceAllString(code, "var ")
+
 	code = strings.ReplaceAll(code, "\r\n", "\n")
-	re = regexp.MustCompile(`(?m)}\n *catch`)
-	code = re.ReplaceAllString(code, "} catch")
-	return code
+
+	lines := strings.Split(code, "\n")
+
+	for _, line := range lines {
+		re := regexp.MustCompile(`(?m)let |const `)
+		line = re.ReplaceAllString(line, "var ")
+
+		if strings.Contains(line, "~") {
+			if !strings.Contains(line, "var ") {
+				line = fmt.Sprintf("var hidden = %s", line)
+			}
+			line = strings.ReplaceAll(line, "~", "")
```

Comment:
Also regex here? `"^(\s*)~"` and substitute with `"$1 "`

Reply:
added in last commit